src/main/java/org/apache/maven/plugins/pmd/PmdReport.java

```diff
@@ -769,7 +769,7 @@ private void configureTypeResolution( PMDConfiguration configuration ) throws Ma
                         for ( String path : projectCompileClasspath )
                         {
                             File pathFile = new File( path );
-                            if ( !pathFile.exists() || pathFile.list().length == 0 )
+                            if ( !pathFile.exists() || pathFile.list() == null || pathFile.list().length == 0 )
```

Comment:
This fix is not correct.
Because the second invocation may return null.
You have to assign the result of list() to a local variable.

Reply:
In not following.

If the second clause is true, the third clause never executes.

If the second clause is false, the third one won't throw an NPE.

Can you suggest code that is better for this?  (Or describe why a local variable is better?)

Reply:
If the first call to list() returns a non-null value the jvm will evaluate the third condition but a new call to list() is not guaranteed to be null. (In real world it won't be null I know)

Tools like spotbugs usually find this kind of potential problems.

I am just nitpicking.

If you can list() only once you won't have problems

Reply:
@eolivelli 
yes, you are correct.
See updates in c64e1f659ce8ace2e646e596acd8d705d367f414

@adangel  
See https://issues.apache.org/jira/browse/MPMD-288

I ran `mvn clean verify` and `mvn -Prun-its clean verify` (both completed successfully)